src/ducks/Signals/signalFormManager/signalCrudForm/signal/TriggerForm.js

```diff
@@ -258,66 +293,158 @@ export const TriggerForm = ({
                           name='timeWindowUnit'
                           className={styles.timeWindowUnit}
                           clearable={false}
                           placeholder='Unit'
                           options={[
-                            { value: 'h', label: 'hours' },
-                            { value: 'd', label: 'days' }
+                            { value: 'h', label: 'Hours' },
+                            { value: 'd', label: 'Days' }
                           ]}
                         />
                       </div>
                     </div>
                   )}
                 </div>
+
                 <div className={styles.row}>
                   <div className={styles.Field}>
-                    <label>Message Frequency</label>
-                    <div>Once per</div>
-                    <div className={styles.Field}>
-                      <FormikSelector
-                        name='cooldown'
-                        options={['1h', '24h']}
-                        defaultSelected={'1h'}
+                    <label>Frequency of notifications</label>
+                    <FormikSelect
+                      name='frequencyType'
+                      isClearable={false}
+                      isDisabled={defaultFrequencyType.isDisabled}
+                      defaultValue={defaultFrequencyType.value.value}
+                      isSearchable
+                      placeholder='Choose frequency'
+                      options={FREQUENCY_TYPES}
+                      onChange={frequencyType => {
+                        const newFrequencyTimeType = getNearestFrequencyTypeValue(
+                          frequencyType
+                        )
+                        setFieldValue('frequencyTimeType', newFrequencyTimeType)
+                        setFieldValue(
+                          'frequencyTimeValue',
+                          getNearestFrequencyTimeValue(newFrequencyTimeType)
+                        )
+                      }}
+                    />
+                  </div>
+                  <div className={styles.Field}>
+                    <label>&nbsp;</label>
+
+                    <div className={styles.frequency}>
+                      <FormikSelect
+                        name='frequencyTimeValue'
+                        className={styles.frequencyTimeValue}
+                        isClearable={false}
+                        isDisabled={
+                          !values.frequencyType || !values.frequencyTimeType
+                        }
+                        isSearchable
+                        options={getFrequencyTimeValues(
+                          values.frequencyTimeType
+                        )}
+                      />
+                      <FormikSelect
+                        className={styles.frequencyTimeType}
+                        name='frequencyTimeType'
+                        isDisabled={!values.frequencyType}
```

Comment:
Please, destructure `values` object. There are multiple occurrences (LOC `439`, `429`, `368` and so on) where its properties are being accessed.

Reply:
Will fixed